builder/dockerfile/instructions/commands.go

```diff
@@ -0,0 +1,381 @@
+package instructions
+
+import (
+	"errors"
+
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/strslice"
+)
+
+// KeyValuePair represent an arbitrary named value (usefull in slice insted of map[string] string to preserve ordering)
+type KeyValuePair struct {
+	Key   string
+	Value string
+}
+
+func (kvp *KeyValuePair) String() string {
+	return kvp.Key + "=" + kvp.Value
+}
+
+// KeyValuePairs is a slice of KeyValuePair
+type KeyValuePairs []KeyValuePair
+
+// BaseCommand is the base class for Dockerfile (String() returns its source code)
+type BaseCommand struct {
```

Comment:
nit: `Base` is not a descriptive name. Maybe just `named` since it's an instruction in the instructions package, and it's not used externally.